src/parse.c

```diff
@@ -1945,12 +1946,59 @@ void ParseDecorations(const TokenNode *tp, DecorationsType *deco)
 /** Parse a color which may be a gradient. */
 void ParseGradient(const char *value, ColorType a, ColorType b)
 {
-
    const char *sep;
    char *temp;
 
    /* Find the separator. */
    sep = strchr(value, ':');
+   
+   if(!sep) {
+
+      /* Only one color given - use the same color for both. */
+      SetColor(a, value);
+      SetColor(b, value);
+
+   } else {
+
+      /* Two colors. */
+
+      /* Get the first color. */
+      int len = (int)(sep - value);
+      temp = AllocateStack(len + 1);
+      memcpy(temp, value, len);
+      temp[len] = 0;
+      SetColor(a, temp);
+      ReleaseStack(temp);
+
+      /* Get the second color. */
+      len = strlen(sep + 1);
+      temp = AllocateStack(len + 1);
+      memcpy(temp, sep + 1, len);
+      temp[len] = 0;
+      SetColor(b, temp);
+      ReleaseStack(temp);
+
+   }
+}
+void ParseGradientType(const char *value, ColorType a, ColorType b, GradientType g)
+{
+
+   const char *sepv;
+   const char *seph;
+   const char *sep;
+   char *temp;
+
+   /* Find the separator. */
+   sepv = strchr(value, ':');
+   seph = strchr(value, ';');
+
+   sep = seph;
+   SetGradient(g, GRADIENT_HORIZONTAL);
+    
+   if (sepv) {
+      sep = sepv;
+      SetGradient(g, GRADIENT_VERTICAL);
+   }
```

Comment:
To remove duplication, I think I would merge `ParseGradient` and `ParseGradientType` and just have a dummy gradient type for when it's not used (maybe use that missing `0` as `GRADIENT_NONE`).

Reply:
Yes this is the first approach I suppose to use but I got some error during the make step and I choose to duplicate the code.
Now I tested a better code with all your suggestions. I will pull it today so, if I found how to, I will remove this pull and create a new one where I also added support for background gradient.